src/coprocessor/endpoint.rs

```diff
@@ -327,8 +316,13 @@ impl RequestTask {
     }
 
     #[inline]
-    fn check_outdated(&self) -> Result<()> {
-        self.ctx.check_if_outdated()
+    fn check_outdated(&self, request_max_handle_duration: Duration) -> Result<()> {
+        let now = Instant::now_coarse();
+        let deadline = self.timer + request_max_handle_duration;
```

Comment:
Why is deadline calculated every time in the new implementation, instead of using pre-calculated "deadline"?

Reply:
Deadline is removed.

Reply:
Maybe you could keep both deadline && start_time in RequestTask, so 
1. you needn't pass request_max_handle_duration as a parameter in so many functions
2. It seems you only need to compute elapsed time in `err_resp`, while you could compute it  by `RequestTask.start_time`

Reply:
2 is fixed. 1 is partially fixed, this is the best I can do, because `request_max_handle_duration` has been removed from `KvService`.